src/CoreWCF.BuildTools/src/OperationParameterInjectionGenerator.cs

```diff
@@ -0,0 +1,235 @@
+            var CoreWCFServiceBehaviorSymbol = context.Compilation.GetTypeByMetadataName("CoreWCF.ServiceBehaviorAttribute");
+
+            string fileName = $"{contract.Name}_{operationContract.Name}.cs";
+            var dependencies = operationContractImplementation.Parameters.Where(x => !operationContract.Parameters.Any(p =>
+                   p.IsMatchingParameter(x))).ToArray();
+
+            bool isInstanceContextModeSingle = GetInstanceContextMode(context, service);
+            
+            var builder = new StringBuilder();
+            builder.Append($@"
+namespace {service.ContainingNamespace}
+{{
+    public partial class {service.Name}
+    {{
+");
+            builder.Append($@"        public {GetReturnType()} {operationContract.Name}({GetParameters()})
+        {{
+");
+            builder.Append($@"            System.IServiceProvider serviceProvider = CoreWCF.OperationContext.Current.InstanceContext.Extensions.Find<System.IServiceProvider>();
+            if (serviceProvider == null) throw new System.InvalidOperationException(""Missing IServiceProvider in InstanceContext extensions"");
+");
+            if (isInstanceContextModeSingle)
+            {
+                builder.Append($@"            using (var scope = serviceProvider.CreateScope())
+            {{
+");
+            }    
+            
+            AddDependenciesResolution();
+            AddMethodCall();
+
+            if (isInstanceContextModeSingle)
+            {
+                builder.Append($@"
+            }}");
+            }
+
+            builder.Append($@"
+        }}
+    }}
+}}
+");
+            context.AddSource(fileName, SourceText.From(builder.ToString(), Encoding.UTF8, SourceHashAlgorithm.Sha256));
+
+            string GetReturnType()
+                => operationContract.ReturnsVoid ?
+                    "void"
+                    : $"{operationContract.ReturnType.ContainingNamespace}.{operationContract.ReturnType.Name}";
+
+            string GetParameters()
+                => string.Join(", ", operationContract.Parameters
+                    .Select(p => $"{p.Type.ContainingNamespace}.{p.Type.Name} {p.Name}"));
+            
+            void AddDependenciesResolution()
+            {
+                string serviceProviderName = isInstanceContextModeSingle ?
+                    "scope"
+                    : "serviceProvider";
+                string prefix =  isInstanceContextModeSingle ?
+                    "                "
+                    : "            ";
+                for (int i = 0; i < dependencies.Length; i++)
+                {
+                    builder.AppendLine($@"{prefix}var d{i} = ({dependencies[i].Type.ContainingNamespace}.{dependencies[i].Type.Name}){serviceProviderName}.GetService(typeof({dependencies[i].Type.ContainingNamespace}.{dependencies[i].Type.Name}));");
+                }
```

Comment:
Is there a reason to use `GetService(typeof(typename))` instead of `GetService<typename>();`?

Reply:
I was trying to avoid output `using` directives in generated code. I’m now using the `GetService<T>()` extension method as I had to add a `using Microsoft.Extensions.DependencyInjection;` to call `CreateScope()`